### PR TITLE
Add draft of development guidelines

### LINKED ISSUE #1302: Create contribution guidelines to help onboard new devs
Create a document indicating what to do to contribute. Something like http://pandas-docs.github.io/pandas-docs-travis/contributing.html

Might contain:

 - Setting up a developing environment.
 - How to find bugs easy to tackle (issue tag).
 - Issue / Branch / Pull-Request cycle. 
 - Test Driven Development, how to write and run tests.
 - How to contribute to docs, compile them etc.
 - Coding standards (style).
 - Where to ask for help developing (mailing lists, gitter, etc.)

### PR DESCRIPTION
**Pull request by mrocklin**
Fixes #1302 

We've had a number of new developers show up in the last few weeks (Hooray!). Development guidelines were requested and seem to be a great idea.  This both helps newcomers and also helps the more established community here to establish shared practices.  

I've written down some brief thoughts in this PR and encourage commentary or direct edits.  I've pushed this directly to the `dask/` fork so that others with commit rights can push directly.

### CODE REVIEW

**Comment by mrocklin** [comment]
cc @dukebody

**Comment by eriknw** [comment]
Do you want to add `CONTRIBUTING.rst` to the project root?  See:

https://help.github.com/articles/setting-guidelines-for-repository-contributors/

File: docs/source/develop.rst [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+   pip install -e .[complete]
+
+
+Run Tests
+~~~~~~~~~
+
+Dask uses py.test_ for testing.  You can run tests from the main dask directory
+as follows::
+
+   py.test dask --verbose
```
Comment (dukebody) [inline docs/source/develop.rst]: py.test is not installed by default when installing dask. We might want to add a specific dev requirements file like https://github.com/pydata/pandas/blob/master/ci/requirements_dev.txt

**Comment by canavandl** [comment]
Bokeh added GitHub templates for new issues and PRs (https://github.com/bokeh/bokeh/tree/master/.github) a few months ago. We've found them to be super valuable in terms of getting more descriptive issues/PR and pushing support questions to support channels. I'd recommend about thinking about doing the same.

File: docs/source/develop.rst [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+   git clone git@github.com:dask/dask.git
+
+
+Install
+~~~~~~~
+
+Install dask and dependencies::
+
+   cd dask
+   pip install -e .[complete]
```
Comment (dukebody) [inline docs/source/develop.rst]: Installing a complete system with numpy and pandas using pip might not be desirable for Windows users. We can instruct users to use a conda environment like in http://pandas.pydata.org/pandas-docs/stable/contributing.html#creating-a-development-environment

File: docs/source/develop.rst [hunk 3]
```diff
@@ -0,0 +1,216 @@
+Development Guidelines
+======================
+
+Dask is a community maintained project.  We welcome contributions in the form
+of bug reports, documentation, code, design proposals, and more.
+This page provides resources to on how best to contribute.
```
Comment (eriknw) [inline docs/source/develop.rst]: "...resources ~~to~~ on how best to contribute." <-- extraneous "to"

File: docs/source/develop.rst [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+Contributing to Code
+--------------------
+
+Dask maintains development standards that are similar to most PyData projects.  These standards include
+language support, testing, documentation, and style.
+
+Python Versions
+~~~~~~~~~~~~~~~
+
+Dask supports Python versions 2.7, 3.3, 3.4, and 3.5 in a single codebase.
```
Comment (dukebody) [inline docs/source/develop.rst]: Perhaps link to `compatibility.py` here.

**Comment by mrocklin** [comment]
Thank you all for the excellent comments.  Please keep 'em coming.

File: docs/source/develop.rst (hunk 4 above)
Comment (mrocklin) [inline docs/source/develop.rst]: Added a mention.

File: docs/source/develop.rst (hunk 3 above)
Comment (mrocklin) [inline docs/source/develop.rst]: Fixed

File: docs/source/develop.rst (hunk 2 above)
Comment (mrocklin) [inline docs/source/develop.rst]: I've added notes about `conda`.  Further review of the language appreciated

File: docs/source/develop.rst (hunk 1 above)
Comment (mrocklin) [inline docs/source/develop.rst]: Added a note about installing `py.test` and `moto` in the docs here.  Haven't yet produced a separate requirements file.  Should we?

**Comment by mrocklin** [comment]
I've added a contributing.md file that just links to the doc page.  I'm not sure yet what we would put into the issue and PR file that would always be useful.

**Comment by mrocklin** [comment]
Merging soon if no comments

**Pull request merged**